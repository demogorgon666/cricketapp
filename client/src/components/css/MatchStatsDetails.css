@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

.top-container-stats{
    display: grid;
    grid-template-columns: 12% auto 12%;
    height: 400px;
    border: 1px solid #272727;
}
.stats-container{
    display: grid;
    grid-template-rows: 10% auto;
    border: 1px solid #272727;
}