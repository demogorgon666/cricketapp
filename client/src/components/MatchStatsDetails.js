import React, { Component } from "react";
import "../components/css/MatchStatsDetails.css";
import {
  getManhattanGraphbyId,
  getPieChartOnebyId,
  getPieChartTwobyId,
} from "../actions/Matches";
import { connect } from "react-redux";
// import Loader from "react-loader-spinner";
// import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

export class MatchStatsDetails extends Component {
  componentDidMount() {
    // [yatin] calling indivisual stats api on component mount
    this.props.getManhattanGraphbyId(this.props.match_id);
    this.props.getPieChartOnebyId(this.props.match_id);
    this.props.getPieChartTwobyId(this.props.match_id);
  }
  render() {
    console.log("received", this.props.match_stats_pie1);
    console.log("received typeof", typeof this.props.match_stats_pie1);
    console.log("accessing head", this.props.match_stats_pie1.head);
    console.log("accessing head2", this.props.match_stats_pie1.head2);
    console.log(
      "accessing status_code",
      this.props.match_stats_pie1.status_code
    );
    console.log("accessing message", this.props.match_stats_pie1.message);
    return (
      <div>
        {/* {this.props.isLoading ? (
          <div style={{ margin: "400px" }}>
            <Loader
              type="TailSpin"
              color="#2980b9"
              height="100"
              width="100"
            />
          </div>
        ) : ( */}
        <div style={{ marginTop: 230 + "px" }}>
          <div className="stats-main-container">
            <div id="manhattan-container" className="stats-container">
              <p id="manhattan-title" className="top-title">
                Manhattan
              </p>
              <div>
                <iframe
                  className="stats-graph"
                  src={this.props.match_stats_manhattan.head}
                />
              </div>
            </div>
            <div className="two-pie-chart-container">
              <div className="stats-container">
                <p className="top-title"> Pie chart1 of players run</p>
                <div>
                  <iframe
                    className="stats-graph"
                    src={this.props.match_stats_pie1.head}
                  />
                </div>
              </div>
              <div className="stats-container">
                <p className="top-title"> Pie chart2 of players run</p>
                <div>
                  <iframe
                    className="stats-graph"
                    src={this.props.match_stats_pie1.head2}
                  />
                </div>
              </div>
            </div>
            <div className="two-pie-chart-container">
              <div className="stats-container">
                <p className="top-title"> Pie1 chart of bowlers wicket</p>
                <div>
                  <iframe
                    className="stats-graph"
                    src={this.props.match_stats_pie2.head}
                  />
                </div>
              </div>
              <div className="stats-container">
                <p className="top-title"> Pie2 chart of bowlers wicket</p>
                <div>
                  <iframe
                    className="stats-graph"
                    src={this.props.match_stats_pie2.head2}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div >
    );
  }
}

const mapStateToProps = state => ({
  match_stats_manhattan: state.matchreducer.match_stats_manhattan,
  match_stats_pie1: state.matchreducer.match_stats_pie1,
  match_stats_pie2: state.matchreducer.match_stats_pie2,
  isLoading: state.LoadingReducer.isLoading,
});

export default connect(
  mapStateToProps,
  { getManhattanGraphbyId, getPieChartOnebyId, getPieChartTwobyId }
)(MatchStatsDetails);
