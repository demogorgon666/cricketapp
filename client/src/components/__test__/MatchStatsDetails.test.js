import React from "react";
import { shallow } from "enzyme";
import MatchStatsDetails from "../MatchStatsDetails"
import { Provider } from "react-redux";
import store from "../../store";

const wrapper = shallow(
    <Provider store={store}>
        <MatchStatsDetails />
    </Provider>
);


describe("Testing of MatchStatsDetails Component", () => {

    it("should render the component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("[Main container] should have exactly defined classname ", () => {
        expect(
            wrapper
                .find("div")
                .at(3)
                .props().className
        ).toBe("top-container-stats");
    });
    it("[Main container -> stats-container] should have exactly defined classname ", () => {
        expect(
            wrapper
                .find("div")
                .at(5)
                .props().className
        ).toBe("stats-container");
    });
    it("[Main container -> top-title] should have exactly defined classname ", () => {
        expect(
            wrapper
                .find("div")
                .at(6)
                .props().className
        ).toBe("top-title");
    });




});
