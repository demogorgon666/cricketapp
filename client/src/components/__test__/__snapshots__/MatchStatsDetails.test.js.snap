// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of MatchScoreDetails Component should render the component 1`] = `
<div>
  <div
    className="main-spacing-container"
  >
    <div
      className="top-container-scorecard"
      style={
        Object {
          "borderRadius": "5px",
        }
      }
    >
      <div
        id="scorecard-inning"
        style={
          Object {
            "backgroundColor": "#2980B9",
            "borderTopLeftRadius": "5px",
            "borderTopRightRadius": "5px",
            "color": "white",
            "fontSize": "400",
            "padding": "10px",
          }
        }
      >
        Inning 
      </div>
      <div
        className="main-container-scorecard"
      >
        <div
          className="batsman-heading"
          style={
            Object {
              "fontWeight": "500",
              "padding": "5px",
            }
          }
        >
          <div
            id="batsmen-text"
          >
            Batsmen
          </div>
          <div />
          <div
            id="runs-text"
          >
            R
          </div>
          <div
            id="ball-text"
          >
            B
          </div>
          <div
            id="fours-text"
          >
            4s
          </div>
          <div
            id="sixes-text"
          >
            6s
          </div>
          <div
            id="SR-text"
          >
            SR
          </div>
        </div>
        <div>
          <div
            className="batsman-heading"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              id="batsmen-name"
            >
              PJ Hughes
            </div>
            <div
              id="how-out"
            >
              (c) 
              ND McKenzie
               
               (b) 
               
              JH Kallis
            </div>
            <div
              id="batsmen-run"
            >
              115
            </div>
            <div
              id="batsmen-ball-faced"
            >
              151
            </div>
            <div
              id="batsmen-fours"
            >
              19
            </div>
            <div
              id="batsmen-sixes"
            >
              2
            </div>
            <div
              id="batsmen-strike-rate"
            >
              76.16
            </div>
          </div>
        </div>
        <div
          className="extras-content"
          style={
            Object {
              "fontWeight": "500",
              "padding": "10px",
            }
          }
        >
          <div
            id="match-extras-text"
          >
            Extras
          </div>
          <div
            id="match-extras-value"
            style={
              Object {
                "margin": "0 80px 0 0",
              }
            }
          >
            15
             extras
          </div>
        </div>
        <div
          className="total-content"
          style={
            Object {
              "borderBottom": "none",
              "fontWeight": "500",
              "padding": "5px",
            }
          }
        >
          <div
            id="total-text"
          >
            Total
          </div>
          <div
            id="total-value"
            style={
              Object {
                "margin": "0 40px 0 0",
              }
            }
          >
            352
             /
            10
             (
            106
             overs)
          </div>
        </div>
        <div
          className="bowler-heading"
          style={
            Object {
              "borderTop": "none",
              "fontWeight": "500",
              "padding": "5px",
            }
          }
        >
          <div
            id="bowler-text"
          >
            Bowler
          </div>
          <div />
          <div
            id="bowler-over-text"
          >
            O
          </div>
          <div
            id="bowler-run-text"
          >
            R
          </div>
          <div
            id="bowler-wicket-text"
          >
            W
          </div>
          <div
            id="bowler-extras-text"
          >
            Extra
          </div>
          <div
            id="bowler-ecom-text"
          >
            Ecom
          </div>
        </div>
        <div>
          <div
            className="bowler-heading"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              id="bowler-name-value"
            >
              M Ntini
            </div>
            <div />
            <div
              id="bowler-overs-value"
            >
              19
            </div>
            <div
              id="bowler-runs-value"
            >
              58
            </div>
            <div
              id="bowler-wicket-value"
            >
              2
            </div>
            <div
              id="bowler-extras-value"
            >
              0
            </div>
            <div
              id="bowler-ecom-value"
            >
              3.05
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing of MatchStatsDetails Component should render the component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(MatchStatsDetails) />
</ContextProvider>
`;
